File: tests/Pieces/IndirectTest/java/AccessTresholdPlusOne/src/MyTest.java
import org.junit.TestCase;

// use treshold = 4

public class MyTest extends TestCase {


    public void testCommand() {
        int i;

        A aa = new A();
        B bb = new B();
        C cc = new C();
        D dd = new D();
        E ee = new E();


        i = aa.a;
        i = bb.b;
        i = cc.c;
        i = dd.d;
        i = ee.e;
    }

}

File: tests/Pieces/IndirectTest/cpp/AccessInvokeUnderTreshold/src/MyTest.cpp
#include "cppunit/TestSuite.h"
#include "cppunit/TestCaller.h"
#include "MyTest.h"
#include "Stuff.h"

using CppUnit::TestSuite;
using CppUnit::TestCaller;

void MyTest::testCommand() {
    A a;
    B b;
    a.a();
    b.b();

    D d;
    int i;
    i = d.d;
}

Test* MyTest::suite() {
    TestSuite *suite = new TestSuite();
    suite->addTest(new TestCaller<MyTest>("testCommand", &MyTest::testCommand));
    return suite;
}
